### PR TITLE
feat(dart): add async/sync* to highlights

### PR DESCRIPTION
**Pull request by akinsho**
Relates to https://github.com/UserNobody14/tree-sitter-dart/issues/18
Not sure what the outcome of that issue will be but this attempts to update/fix some highlights in the interim.

### Todo
- [x] fix async/sync*
- [x] Update type regex to only match Uppercase classes not CONSTANTS

### CODE REVIEW

File: queries/dart/highlights.scm [hunk 1]
```diff
@@ -205,7 +208,7 @@
 
 ; when used as an identifier:
 ((identifier) @variable.builtin
- (#vim-match? @variable.builtin "^(abstract|as|covariant|deferred|dynamic|export|external|factory|Function|get|implements|import|interface|library|operator|mixin|part|set|static|typedef)$"))
+ (#match? @variable.builtin "^(abstract|as|covariant|deferred|dynamic|export|external|factory|Function|get|implements|import|interface|library|operator|mixin|part|set|static|typedef)$"))
```
Comment (theHamsta) [inline queries/dart/highlights.scm:211]: You could switch this to use `any-of?`. `#any-of? @variable.builtin "abstract" "as" ...`

**Review by vigoux — APPROVED** [review]

File: queries/dart/highlights.scm [hunk 2]
```diff
@@ -105,13 +105,11 @@
 (final_builtin) @constant.builtin
 
 ((identifier) @type
- (#match? @type "^_?[A-Z]"))
+ (#match? @type "^_?[A-Z][a-z]"))
```
Comment (theHamsta) [inline queries/dart/highlights.scm:108]: I used `[A-Z].*[a-z]` in one of the other languages to also catch stuff like `IBlock` and `VBuffer`

Comment (akinsho) [reply queries/dart/highlights.scm]: Thanks for the pointer just tweaked it

**Comment by theHamsta** [comment]
Thank you!

File: queries/dart/highlights.scm [hunk 3]
```diff
@@ -90,7 +87,6 @@
   name: (identifier) @type)
 (enum_constant
   name: (identifier) @type)
-(type_identifier) @type
```
Comment (theHamsta) [inline queries/dart/highlights.scm:93]: Why is `type_identifier` not needed anymore?

Comment (akinsho) [reply queries/dart/highlights.scm:93]: It was a duplicate, the exact same thing is present at line 97. There was a bit of messiness generally so some of the other changes are tidying up.

**Comment by akinsho** [comment]
Sorry for that last force push @theHamsta there were some typos in the original commit so just wanted to fix those up

**Pull request merged**